server/datastore/cached_mysql/cached_mysql_test.go

```diff
@@ -13,6 +14,54 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestClone(t *testing.T) {
+	tests := []struct {
+		name string
+		src  interface{}
+		want interface{}
+	}{
+		{
+			name: "string",
+			src:  "foo",
+			want: "foo",
+		},
+		{
+			name: "struct",
+			src: fleet.AppConfig{
+				ServerSettings: fleet.ServerSettings{
+					EnableAnalytics: true,
+				},
+			},
+			want: fleet.AppConfig{
+				ServerSettings: fleet.ServerSettings{
+					EnableAnalytics: true,
+				},
+			},
+		},
+		{
+			name: "pointer to struct",
+			src: &fleet.AppConfig{
+				ServerSettings: fleet.ServerSettings{
+					EnableAnalytics: true,
+				},
+			},
+			want: &fleet.AppConfig{
+				ServerSettings: fleet.ServerSettings{
+					EnableAnalytics: true,
+				},
+			},
```

Comment:
Hmm, I don't want to add one for each type specifically, as that doesn't add much value IMO. Instead, it would be good to add a test for each "type of type" like slices, maps, pointers to slices, etc.

Reply:
I was just thinking it as a safe check with the types defined this package.

> Instead, it would be good to add a test for each "type of type" like slices, maps, pointers to slices, etc.

Ya, that sounds more of a test for `copier` functionality.

So, I think we are good then.

Reply:
Added tests